rosetta/transactions/parse_transaction.go

```diff
@@ -15,16 +15,208 @@
+	sender := identifier.Account{
+		Address: authorizer.String(),
+	}
+
+	// Validate the sender address.
+	err := p.validate.Account(sender)
+	if err != nil {
+		return nil, nil, fmt.Errorf("invalid sender account: %w", err)
+	}
+
+	// Verify that the sender is the payer and the proposer.
+	if tx.Payer != authorizer {
+		return nil, nil, failure.InvalidPayer{
+			Have:        tx.Payer,
+			Want:        authorizer,
+			Description: failure.NewDescription("invalid transaction payer"),
+		}
+	}
+	if tx.ProposalKey.Address != authorizer {
+		return nil, nil, failure.InvalidProposer{
+			Have:        tx.ProposalKey.Address,
+			Want:        authorizer,
+			Description: failure.NewDescription("invalid transaction proposer"),
+		}
+	}
+
+	// Verify the transaction script is the token transfer script.
+	script, err := p.generate.TransferTokens(dps.FlowSymbol)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not generate transfer script: %w", err)
+	}
+	if !bytes.Equal(script, tx.Script) {
+		return nil, nil, failure.InvalidScript{
+			Description: failure.NewDescription("transaction text is not valid token transfer script"),
+		}
+	}
+
+	// Verify that the transaction script has the correct number of arguments.
+	args := tx.Arguments
+	if len(args) != argsRequired {
+		return nil, nil, failure.InvalidArguments{
+			Have:        uint(len(args)),
+			Want:        argsRequired,
+			Description: failure.NewDescription("invalid number of arguments"),
+		}
+	}
+
+	// Parse and validate the amount argument.
+	val, err := json.Decode(args[0])
+	if err != nil {
+		return nil, nil, failure.InvalidAmount{
+			RawAmount: string(args[0]),
+			Description: failure.NewDescription("could not parse transaction amount",
+				failure.WithErr(err)),
+		}
+	}
+	amountArg, ok := val.ToGoValue().(uint64)
+	if !ok {
+		return nil, nil, failure.InvalidAmount{
+			RawAmount:   string(args[0]),
+			Description: failure.NewDescription("invalid amount"),
+		}
+	}
+	amount := strconv.FormatUint(amountArg, 10)
+
+	// Parse and validate receiver script argument.
+	val, err = json.Decode(args[1])
+	if err != nil {
+		return nil, nil, failure.InvalidReceiver{
+			RawReceiver: string(args[1]),
+			Description: failure.NewDescription("could not parse transaction receiver address",
+				failure.WithErr(err)),
+		}
+	}
+	receiver := identifier.Account{
+		Address: val.String(),
+	}
+	err = p.validate.Account(receiver)
+	if err != nil {
+		return nil, nil, fmt.Errorf("invalid receiver account: %w", err)
+	}
+
+	// Validate the reference block identifier.
+	rosBlockID := identifier.Block{
+		Hash: tx.ReferenceBlockID.String(),
+	}
+	_, err = p.validate.Block(rosBlockID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("invalid reference block: %w", err)
+	}
+
+	// Create the send operation.
+	sendOp := object.Operation{
+		ID: identifier.Operation{
+			Index: 0,
+		},
+		RelatedIDs: []identifier.Operation{{Index: 1}},
+		AccountID:  sender,
+		Type:       dps.OperationTransfer,
```

Comment:
In the future, we might want to match the script against a type in some kind of registry, but yes, it's fine to do it like this now.